### PR TITLE
Fixed #1 -- do stuff concurrently

### LINKED ISSUE #1: Call Service.get_user concurrently in main

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -54,9 +64,11 @@ fn main() {
             };
             println!("@{}{}{}", user.name, email, details);
         }
-        if i + 1 != services.len() {
```
Comment (frewsxcv) [inline src/main.rs]: Alternatively

```rust
if count == i + 1 {
    break;
}
println!("");
println!("");
```

**Comment by alex** [comment]
Compilation failing with:

```
~/p/h/otp-cop (concurrent) $ cargo build
   Compiling otp-cop v0.1.0 (file:///Users/alex_gaynor/projects/hacks/otp-cop)
src/main.rs:34:22: 34:35 error: the parameter type `U` may not live long enough [E0310]
src/main.rs:34         threads.push(thread::spawn(move || {
                                    ^~~~~~~~~~~~~
note: in expansion of for loop expansion
src/main.rs:33:5: 37:6 note: expansion site
src/main.rs:34:22: 34:35 help: run `rustc --explain E0310` to see a detailed explanation
src/main.rs:34:22: 34:35 help: consider adding an explicit lifetime bound `U: 'static`...
src/main.rs:34:22: 34:35 note: ...so that captured variable `tx` does not outlive the enclosing closure
src/main.rs:34         threads.push(thread::spawn(move || {
                                    ^~~~~~~~~~~~~
note: in expansion of for loop expansion
src/main.rs:33:5: 37:6 note: expansion site
src/main.rs:34:22: 34:35 error: the parameter type `T` may not live long enough [E0310]
src/main.rs:34         threads.push(thread::spawn(move || {
                                    ^~~~~~~~~~~~~
note: in expansion of for loop expansion
src/main.rs:33:5: 37:6 note: expansion site
src/main.rs:34:22: 34:35 help: run `rustc --explain E0310` to see a detailed explanation
src/main.rs:34:22: 34:35 help: consider adding an explicit lifetime bound `T: 'static`...
src/main.rs:34:22: 34:35 note: ...so that captured variable `o` does not outlive the enclosing closure
src/main.rs:34         threads.push(thread::spawn(move || {
                                    ^~~~~~~~~~~~~
note: in expansion of for loop expansion
src/main.rs:33:5: 37:6 note: expansion site
error: aborting due to 2 previous errors
Could not compile `otp-cop`.

To learn more, run the command again with --verbose.
```

File: src/main.rs [hunk 2]
```diff
@@ -2,11 +2,48 @@ extern crate getopts;
...
+        where F1: 'static + Fn(T) -> U + Send, T: Send, U: Send {
+    let (tx, rx) = mpsc::channel();
+    let mut threads = vec![];
+    for o in objs {
+        threads.push(thread::spawn(move || {
+            tx.send(f1(o));
+        }));
+    }
+
+    for t in threads {
```
Comment (nelhage) [inline src/main.rs]: I feel like ideally you don't need to join all the threads here – there's no reason the client can't consume as work comes in. You can presumably tie the lifetimes to the lifetime of the `ParallelIter` to make sure things get cleaned up, somehow.

**Comment by alex** [comment]
New error message:

```
~/p/h/otp-cop (concurrent) $ cargo build
   Compiling otp-cop v0.1.0 (file:///Users/alex_gaynor/projects/hacks/otp-cop)
src/main.rs:37:21: 37:23 error: capture of moved value: `f1` [E0382]
src/main.rs:37             tx.send(f1(o));
                                   ^~
note: in expansion of closure expansion
src/main.rs:36:23: 38:10 note: expansion site
note: in expansion of for loop expansion
src/main.rs:34:5: 39:6 note: expansion site
src/main.rs:36:23: 38:10 note: `f1` moved into closure environment here because it has type `[closure(())]`, which is non-copyable
src/main.rs:36         thread::spawn(move || {
src/main.rs:37             tx.send(f1(o));
src/main.rs:38         });
note: in expansion of closure expansion
src/main.rs:36:23: 38:10 note: expansion site
note: in expansion of for loop expansion
src/main.rs:34:5: 39:6 note: expansion site
src/main.rs:36:23: 38:10 help: perhaps you meant to use `clone()`?
error: aborting due to previous error
Could not compile `otp-cop`.

To learn more, run the command again with --verbose.
```

**Pull request merged**